src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -53,17 +53,30 @@
 #define GPIO_INVALID_MODE 0xE
 #define INVALID_GPIO_OSPEED 0xE
 
-#define HIGH_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\x4", "high"))
-#define LOW_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\x3", "low"))
-
-#define INVALID_BANK_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xC", "invalid_bank"))
-#define INVALID_PIN_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xB", "invalid_pin"))
-#define INVALID_MODE_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xC", "invalid_mode"))
-#define INVALID_PULL_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xC", "invalid_pull"))
-#define INVALID_RATE_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xC", "invalid_rate"))
-#define INVALID_LEVEL_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xD", "invalid_level"))
-#define INVALID_IRQ_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xB", "invalid_irq"))
-#define INVALID_TRIGGER_ATOM globalcontext_make_atom(ctx->global, ATOM_STR("\xF", "invalid_trigger"))
+#define VALIDATE_BANK_VALUE(value)                                                          \
```

Comment:
Perhaps to mitigate @bettio ‘s concerns with these macros we could make the global parameter to the globalcontext function a parameter to the macro.  That way, we are not relying on `ctx` being in scope when the macro is called (which I believe was one of @bettio ’s concerns)

Reply:
That makes sense, in fact I had originally written theses that way, and placed them in the header file thinking that they could be used by future drivers (SPI, I2c, etc...). I really don't know why I decided to change that - perhaps the same concern as the use of macros to create atoms as the pattern is very similar.